### PR TITLE
[COM-249] Create composite key for registry Map

### PR DESCRIPTION
**Pull request by alxgmpr**
Previously, the registry only used the added path as the key to the `Map`. This breaks when you add different middlewares for different methods, like:

```typescript
this.add('/api/abc', middlewareA(), { methods: ['GET'] });
this.add('/api/abc', middlewareB(), { methods: ['PUT'] });
```

Since maps cannot have duplicate keys, only the middleware corresponding to PUT would run because it was the last to be added to the map.

Thus this adds the ability to serialize `config.methods` into the key of the map, and then deserialize the path out of the key whenever we actually need to test the RegEx.

### CODE REVIEW

File: src/MiddlewareRegistry.ts [hunk 1]
```diff
@@ -81,4 +82,24 @@ export class MiddlewareRegistry<R extends MiddlewareRequest> {
...
+    private serializeToRegistryKey(route: string, methods?: MiddlewareConfig['methods']): string {
+        return JSON.stringify({ route, methods });
+    }
+
+    /**
+     * Given a serialized key from the middleware mapping, return the route
+     * @param serializedKey The serialized key to parse
+     * @private
+     */
+    private deserializeKeyToPathAndMethods(serializedKey: string): string {
```
Comment (joshrathke) [inline src/MiddlewareRegistry.ts]: `deserializeKeyToRoute` maybe?

**Review by joshrathke — APPROVED** [review]

File: src/MiddlewareRegistry.ts (hunk 1 above)
Comment (alxgmpr) [reply src/MiddlewareRegistry.ts]: Ah yes, remnant of an earlier version. Will do, thanks.

File: src/MiddlewareRegistry.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ export class MiddlewareRegistry<R extends MiddlewareRequest> {
         middleware: MiddlewareFunction | MiddlewareFunction[],
         config?: Omit<MiddlewareConfig, 'middleware'>
     ) {
-        this.registry.set(route, { ...config, middleware: middleware });
+        this.registry.set(this.serializeToRegistryKey(route, config?.methods), { ...config, middleware: middleware });
```
Comment (joshrathke) [inline src/MiddlewareRegistry.ts]: Nit: Unrelated to this MR, but I think we can reduce `middleware: middleware` to just `middleware`
```suggestion
        this.registry.set(this.serializeToRegistryKey(route, config?.methods), { ...config, middleware });
```

Comment (alxgmpr) [reply src/MiddlewareRegistry.ts]: Good call. Done.

**Review by trynow-david — APPROVED** [review]

**Pull request merged**